requirements-test.txt

```diff
@@ -1,6 +1,6 @@
 jsonschema==4.7.2
 mock==1.3.0
-pytest==6.2.5
-coverage==5.5
-pytest-cov==2.12.1
-pytest-xdist==2.4.0
+pytest==7.2.0
+coverage==7.0.1
+pytest-cov==4.0.0
+pytest-xdist==3.1.0
```

Comment:
We do not need to do this as part of this PR, but we should enable dependabot for these dependencies so that we get auto-updates for them. Here is an example of where we've enabled dependabot for a dependency before: https://github.com/aws/aws-cli/pull/7521

Reply:
This will need to be a prerequisite to this PR if not done here. I believe there are some incompatibilities with older versions of pytest and 3.11.

https://github.com/pytest-dev/py/issues/273
https://github.com/pytest-dev/pytest/issues/9181